l1-contracts/test/NewOutbox.t.sol

```diff
@@ -0,0 +1,237 @@
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    (bytes32[] memory path,) = tree.computeSiblingPath(0);
+    outbox.consume(1, 0, fakeMessage, path);
+    vm.expectRevert(abi.encodeWithSelector(Errors.Outbox__AlreadyNullified.selector, 1, 0));
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testRevertIfPathHeightMismatch() public {
+    DataStructures.L2ToL1Msg memory fakeMessage = _fakeMessage(address(this));
+    bytes32 leaf = fakeMessage.sha256ToField();
+
+    NaiveMerkle tree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    NaiveMerkle biggerTree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT + 1);
+    tree.insertLeaf(leaf);
+
+    (bytes32[] memory path,) = biggerTree.computeSiblingPath(0);
+    vm.expectRevert(
+      abi.encodeWithSelector(
+        Errors.Outbox__InvalidPathLength.selector, _DEFAULT_TREE_HEIGHT, _DEFAULT_TREE_HEIGHT + 1
+      )
+    );
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testRevertIfTryingToConsumeMessageNotInTree() public {
+    DataStructures.L2ToL1Msg memory fakeMessage = _fakeMessage(address(this));
+    bytes32 leaf = fakeMessage.sha256ToField();
+    fakeMessage.content = bytes32(uint256(42069));
+    bytes32 modifiedLeaf = fakeMessage.sha256ToField();
+
+    NaiveMerkle tree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    NaiveMerkle modifiedTree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    modifiedTree.insertLeaf(modifiedLeaf);
+    bytes32 modifiedRoot = modifiedTree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    (bytes32[] memory path,) = modifiedTree.computeSiblingPath(0);
+
+    vm.expectRevert(
+      abi.encodeWithSelector(Errors.MerkleLib__InvalidRoot.selector, root, modifiedRoot)
+    );
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  function testValidInsertAndConsume() public {
+    DataStructures.L2ToL1Msg memory fakeMessage = _fakeMessage(address(this));
+    bytes32 leaf = fakeMessage.sha256ToField();
+
+    NaiveMerkle tree = new NaiveMerkle(_DEFAULT_TREE_HEIGHT);
+    tree.insertLeaf(leaf);
+    bytes32 root = tree.computeRoot();
+
+    vm.prank(_STATE_TRANSITIONER);
+    outbox.insert(1, root, _DEFAULT_TREE_HEIGHT);
+
+    (bytes32[] memory path,) = tree.computeSiblingPath(0);
+
+    vm.expectEmit(true, true, true, true, address(outbox));
+    emit INewOutbox.MessageConsumed(1, root, leaf, 0);
+    outbox.consume(1, 0, fakeMessage, path);
+  }
+
+  // This test takes awhile so to keep it somewhat reasonable we've set a limit on the amount of fuzz runs
+  /// forge-config: default.fuzz.runs = 32
```

Comment:
Is this still true after the removal of the `vm.assume` calls?

Reply:
Can you make the `recipients` array fixed-size, and then pick a a random length using a `size` variable that we bound? Otherwise we don't really know why many recipients we're testing for.